File: backend/src/main/java/by/minilooth/diploma/service/cart/CartService.java
package by.minilooth.diploma.service.cart;

import by.minilooth.diploma.exception.ActionIsImpossibleException;
import by.minilooth.diploma.exception.cart.CartItemNotFoundException;
import by.minilooth.diploma.exception.spareparts.AvailabilityNotFoundException;
import by.minilooth.diploma.exception.spareparts.SparePartNotFoundException;
import by.minilooth.diploma.exception.stores.StoreNotFoundException;
import by.minilooth.diploma.models.bean.cart.Cart;
import by.minilooth.diploma.models.bean.spareparts.SparePart;

import java.util.Collection;
import java.util.List;

public interface CartService {

    void save(Cart cart);
    void save(Collection<Cart> carts);
    List<Cart> getAllBySparePart(SparePart sparePart);

    Cart get();
    Cart add(Long id, Long quantity, Long storeId) throws SparePartNotFoundException, StoreNotFoundException,
            ActionIsImpossibleException;
    Cart delete(Long id) throws SparePartNotFoundException;
    Cart reset();
    Cart increase(Long id) throws SparePartNotFoundException, ActionIsImpossibleException, CartItemNotFoundException,
            AvailabilityNotFoundException;
    Cart decrease(Long id) throws SparePartNotFoundException, ActionIsImpossibleException, CartItemNotFoundException,
            AvailabilityNotFoundException;
    Cart arrange() throws ActionIsImpossibleException, AvailabilityNotFoundException, SparePartNotFoundException,
            StoreNotFoundException;

}

File: frontend/service/cart/CartService.ts
import {Cart} from "types/cart";
import {Axios} from "core/axios";

enum CartRoutes {
  ADD = '/add',
  DELETE = '/delete',
  INCREASE = '/increase',
  DECREASE = '/decrease',
  RESET = '/reset',
  ARRANGE = '/arrange'
}

export class CartService {

  private static readonly URL = '/cart';

  static async get(): Promise<Cart> {
    const {data: cart} = await Axios.get<Cart>(this.URL);
    return cart;
  }

  static async add(sparePartId: number, storeId: number, quantity: number): Promise<Cart> {
    const {data: cart} = await Axios.post<Cart>(this.URL + CartRoutes.ADD, null, {
      params: {
        id: sparePartId,
        storeId,
        quantity
      }
    })
    return cart;
  }

  static async delete(sparePartId: number): Promise<Cart> {
    const {data: cart} = await Axios.post<Cart>(this.URL + CartRoutes.DELETE, null, {
      params: {
        id: sparePartId
      }
    })
    return cart;
  }

  static async increase(sparePartId: number): Promise<Cart> {
    const {data: cart} = await Axios.post<Cart>(this.URL + CartRoutes.INCREASE, null, {
      params: {
        id: sparePartId
      }
    })
    return cart;
  }

  static async decrease(sparePartId: number): Promise<Cart> {
    const {data: cart} = await Axios.post<Cart>(this.URL + CartRoutes.DECREASE, null, {
      params: {
        id: sparePartId
      }
    })
    return cart;
  }

  static async reset(): Promise<Cart> {
    const {data: cart} = await Axios.post<Cart>(this.URL + CartRoutes.RESET, null);
    return cart;
  }

  static async arrange(): Promise<Cart> {
    const {data: cart} = await Axios.post<Cart>(this.URL + CartRoutes.ARRANGE, null);
    return cart;
  }

}
